module/BingoArea.cpp

```diff
@@ -0,0 +1,218 @@
+        id = CIRCLE_ID::ID0;
+      } else if(i == 1 && j == 3) {
+        initCircleColor = COLOR::GREEN;
+        id = CIRCLE_ID::ID1;
+      } else if(i == 1 && j == 5) {
+        initCircleColor = COLOR::RED;
+        id = CIRCLE_ID::ID2;
+      } else if(i == 3 && j == 1) {
+        initCircleColor = COLOR::BLUE;
+        id = CIRCLE_ID::ID3;
+      } else if(i == 3 && j == 5) {
+        initCircleColor = COLOR::YELLOW;
+        id = CIRCLE_ID::ID4;
+      } else if(i == 5 && j == 1) {
+        initCircleColor = COLOR::GREEN;
+        id = CIRCLE_ID::ID5;
+      } else if(i == 5 && j == 3) {
+        initCircleColor = COLOR::RED;
+        id = CIRCLE_ID::ID6;
+      } else if(i == 5 && j == 5) {
+        initCircleColor = COLOR::BLUE;
+        id = CIRCLE_ID::ID7;
+      } else {
+        continue;
+      }
+      blockCircle[index2].setCoordinate(i, j);
+      blockCircle[index2].setCicleColor(initCircleColor);
+      blockCircle[index2].setCircleId(id);
+      index2++;
+    }
+  }
+  centerMark.setCoordinate(3, 3);
+  centerMark.setCicleColor(COLOR::BLACK);
+}
+
+void BingoArea::setBlockInfo(int info, COLOR color)
+{
+  int index = 0;
+  BLOCK_ID blockId = BLOCK_ID::ID0;
+  if(info >= 65 && info <= 83 && info != 73 && info != 78 && info != 79) {
+    //交点サークルの場合
+    switch(info) {
+      case 65:  // Aのとき
+        index = IS_LEFT_COURSE ? 0 : 3;
+        break;
+      case 66:  // Bのとき
+        index = IS_LEFT_COURSE ? 1 : 2;
+        blockId = BLOCK_ID ::ID0;
+        break;
+      case 67:  // Cのとき
+        index = IS_LEFT_COURSE ? 2 : 1;
+        break;
+      case 68:  // Dのとき
+        index = IS_LEFT_COURSE ? 3 : 0;
+        blockId = BLOCK_ID ::ID1;
+        break;
+      case 69:  // Eのとき
+        index = IS_LEFT_COURSE ? 4 : 7;
+        blockId = BLOCK_ID ::ID2;
+        break;
+      case 70:  // Fのとき
+        index = IS_LEFT_COURSE ? 5 : 6;
+        break;
+      case 71:  // Gのとき
+        index = IS_LEFT_COURSE ? 6 : 5;
+        blockId = BLOCK_ID ::ID3;
+        break;
+      case 72:  // Hのとき
+        index = IS_LEFT_COURSE ? 7 : 4;
+        break;
+      case 74:  // Jのとき
+        index = IS_LEFT_COURSE ? 8 : 11;
+        break;
+      case 75:  // Kのとき
+        index = IS_LEFT_COURSE ? 9 : 10;
+        blockId = BLOCK_ID ::ID4;
+        break;
+      case 76:  // Lのとき
+        index = IS_LEFT_COURSE ? 10 : 9;
+        break;
+      case 77:  // Mのとき
+        index = IS_LEFT_COURSE ? 11 : 8;
+        blockId = BLOCK_ID ::ID5;
+        break;
+      case 80:  // Pのとき
+        index = IS_LEFT_COURSE ? 12 : 15;
+        blockId = BLOCK_ID ::ID6;
+        break;
+      case 81:  // Qのとき
+        index = IS_LEFT_COURSE ? 13 : 14;
+        break;
+      case 82:  // Rのとき
+        index = IS_LEFT_COURSE ? 14 : 13;
+        blockId = BLOCK_ID ::ID7;
+        break;
+      case 83:  // Sのとき
+        index = IS_LEFT_COURSE ? 15 : 12;
+        break;
+      default:
+        index = -1;
+    }

```

Comment:
ここの各case文の中で三項演算子書いてるけど，infoが決まってればIS_LEFT_COURSEは一意に決まるよね
（infoが65のとき，IS_LEFT_COURSEは必ず偽である等）
これはなんかキレイに書けそうな気がするけどちょっと解決案は思いつかない

Reply:
```suggestion
    int offset = 65;
    std::array<int, 19> leftIndexTable
        = {0, 1, 2, 3, 4, 5, 6, 7, -1,  8,  9, 10, 11, -1, -1, 12, 13, 14, 15};
    //     A, B, C, D, E, F, G, H,  _,  J,  K,  L,  M,  _,  _,  P,  Q,  R,  S

    std::array<int, 19> rightIndexTable 
        = {3, 2, 1, 0, 7, 6, 5, 4, -1, 11, 10,  9,  8, -1, -1, 15, 14, 13, 12};
    //     A, B, C, D, E, F, G, H,  _,  J,  K,  L,  M,  _,  _,  P,  Q,  R,  S

    std::array<BLOCK_ID, 19> blockIdTable
        = {BLOCK_ID::ID0, BLOCK_ID::ID0, BLOCK_ID::ID0, BLOCK_ID::ID1, BLOCK_ID::ID2, 
          BLOCK_ID::ID0, BLOCK_ID::ID3, BLOCK_ID::ID0, BLOCK_ID::ID0, BLOCK_ID::ID0, 
          BLOCK_ID::ID4, BLOCK_ID::ID0, BLOCK_ID::ID5, BLOCK_ID::ID0, BLOCK_ID::ID0, 
          BLOCK_ID::ID6, BLOCK_ID::ID0, BLOCK_ID::ID7};

    if (IS_LEFT_COURSE ) {
        index = leftIndexTable[info - offset];
    } else {
        index = rightIndexTable[info - offset];
    }
    blockId = blockIdTable[info - offset];
```

個人的に色々試してみたけど、既にあるswitch文と変換テーブルを作成する以外にまともなものが思いつかなかった...

自分は、変換テーブルよりもswitch文のほうが見やすいから、このままのほうがよいと思います。